OpenRA.Mods.Common/Traits/Multipliers/UpgradableMultiplierTrait.cs

```diff
@@ -0,0 +1,75 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public abstract class UpgradeMultiplierTraitInfo
+	{
+		[Desc("Accepted upgrade types.")]
+		public readonly string[] UpgradeTypes = { };
```

Comment:
This needs a `[UpgradeGrantedReference]` I guess since we have https://github.com/OpenRA/OpenRA/pull/8321/ now.